### PR TITLE
update qualimap to 2.3

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by mashehu — APPROVED** [review]

**Comment by maxulysse** [comment]
The mulled container for bamqccram update will be done in a separate PR, as I really just wanted to update the rnaseq module, but ended up doing a tiny bit more than planned

File: modules/nf-core/qualimap/bamqc/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                ]
+                input[1] = []
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert path(process.out.results.get(0).get(1)).exists() },
```
Comment (mashehu) [inline modules/nf-core/qualimap/bamqc/tests/main.nf.test:31]: in the pytest tests there a lot more outputs were snapped. can you please reproduce the pytests.

File: modules/nf-core/qualimap/bamqccram/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                input[2] = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true) ]
+                input[3] = [ file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true) ]
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert path(process.out.results.get(0).get(1)).exists() },
```
Comment (mashehu) [inline modules/nf-core/qualimap/bamqccram/tests/main.nf.test:34]: same here

**Review by mashehu — CHANGES_REQUESTED** [review]
the new snapshots only capture a subset of the pytests. ⛔

File: modules/nf-core/qualimap/bamqc/tests/main.nf.test (hunk 1 above)
Comment (maxulysse) [reply modules/nf-core/qualimap/bamqc/tests/main.nf.test:31]: I copy pasted the logic from the rnaseq tests, but I'll have a look, thanks

Comment (maxulysse) [reply modules/nf-core/qualimap/bamqc/tests/main.nf.test:31]: I probably have myself to blame for that, so I won't even `git blame` to discover that it was me all along

Comment (maxulysse) [reply modules/nf-core/qualimap/bamqc/tests/main.nf.test:31]: ok, so actually I am indeed to blame for the qualimap/rnaseq tests, but they were minimalist

Comment (maxulysse) [reply modules/nf-core/qualimap/bamqc/tests/main.nf.test]: so, so I'm not going to check all the files from the pytest, that was overboard, I think we only need to test the existence of the html file and the snapshot of the txt file

**Review by mashehu — APPROVED** [review]

**Pull request merged**